subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/DefaultArtifactHandler.java

```diff
@@ -55,20 +54,16 @@ private PublishArtifact pushArtifact(org.gradle.api.artifacts.Configuration conf
     }
 
     private PublishArtifact pushArtifact(Configuration configuration, Object notation, Action<? super ConfigurablePublishArtifact> configureAction) {
-        warnIfConfigurationIsDeprecated((DeprecatableConfiguration) configuration);
+        assertConfigurationIsDeclarable(configuration);
         ConfigurablePublishArtifact publishArtifact = publishArtifactFactory.parseNotation(notation);
         configuration.getArtifacts().add(publishArtifact);
         configureAction.execute(publishArtifact);
         return publishArtifact;
     }
 
-    private void warnIfConfigurationIsDeprecated(DeprecatableConfiguration configuration) {
-        if (configuration.isFullyDeprecated()) {
```

Comment:
Same as above. We still support a configuration which is deprecated for declaration. This check should stick around. (in addition to the new error case which was added)